src/cider/nrepl/middleware/info.clj

```diff
@@ -57,9 +59,44 @@
           blacklist
           util/transform-value))))
 
+(defn extract-class-from-compliment [ns-str context]
+  (when (and (seq ns-str)
+             (seq context))
+    (try
+      (when-let [ns-obj (find-ns (symbol ns-str))]
+        (let [c (compliment.context/cache-context context)
+              ^Class c (compliment.sources.class-members/try-get-object-class ns-obj c)]
+          (some-> c .getName)))
+      (catch Exception _
+        ;; We can always be analyzing a broken context.
+        nil))))
+
 (defn info
-  [{:keys [ns sym symbol class member] :as msg}]
-  (let [[ns sym class member] (map misc/as-sym [ns (or sym symbol) class member])
+  [{:keys [ns sym class member context]
+    legacy-sym :symbol
+    :as msg}]
+  (let [sym (or (not-empty legacy-sym)
+                (not-empty sym))
+        class (try
+                (or (when (and (seq class)
```

Comment:
Is this nil-punning here? I still think it doesn't read very well. :D

Reply:
My understanding is that you'd prefer...?

```clj
(when (and (seq class) (seq ns))
  (find-ns (symbol ns)))
```

can do!

Reply:
Not sure of what the alternative would be

(I had written a suggestion, but it wasn't accurate)

Reply:
I'm fine either way - I just don't like much mixing the use of `seq` and `not-empty`. Rich liked `seq`, but `not-empty` was other later on for those who felt `seq` wasn't very intention revealing.

Reply:
(my main point was the one I made earlier - that probably checking for empty strings is redundant, although I understand your reasoning as well)

Reply:
A nice convention is to exclusively use `seq` as a condition, and `not-empty` as an object one intends to further use